processing/src/main/java/org/apache/druid/query/groupby/epinephelinae/LimitedBufferHashGrouper.java

```diff
@@ -318,6 +336,8 @@ public void close()
   private CloseableIterator<Entry<KeyType>> makeHeapIterator()
```

Comment:
Please add some unit tests in the `LimitedBufferHashGrouperTest` where an iterator is created multiple times and still returns the same results.

Reply:
Done in next commit